### ISSUE TITLE
Expose primary renderer store to additional renderer threads

### ISSUE THREAD

**Issue by mtompkins** [chentsulin/electron-react-boilerplate#1098]
Having successfully implemented a 2nd renderer to the boilerplate thanks to @amilajack ( #1095 ) I'm now looking to expose the primary renderer create `store` to this thread. I've tried exporting the `store` variable in various places with no success.

A bit of a push is appreciated. I'm hopeful/sure others will find this small addition helpful as well.

**Comment by amilajack** [comment]
Have you seen this issue? https://github.com/chentsulin/electron-react-boilerplate/issues/108

What was the issue that you were facing? Were you not able to convert your store to a JSON object?

**Comment by mtompkins** [comment]
I've seen this thread, thank you, but haven't moved forward with it because of a number of concerns:
- it appears dated, especially the example timesheet app
- it indicates there may be issues with thunk ("redux-thunk is not FSA-compliant out of the box...")
- it seems primarily based on main to renderer communication, not sharing a single renderer store

... so I've not tried to adapt this approach as it doesn't seem like the correct choice. If I've missed something I'm happy to revisit.

I've been searching for some sort of example purely on the renderer side and to date haven't been successful in finding same.

**Comment by amilajack** [comment]
Have you looked into electron's IPC? You might want to build a wrapper around your store using IPC

**Comment by mtompkins** [comment]
Thank you, however, the issue isn't the interprocess communcation but having a single (or likely sync'd) store to overcome a given threads saturation. Conceptually [electron-redux](https://github.com/hardchor/electron-redux) or similar will be the road foward I've just been unable to integrate it into the boilerplate so far.

**Comment by amilajack** [comment]
@mtompkins what issues did you run into?

**Comment by mtompkins** [comment]
Thanks for asking @amilajack  ... what I'm going to do is spin up the current version of the boilerplate with an added background worker as per the other thread. From there I'll try an implementation again and welcome anyone else who wants to help can offer a PR...that way once finished you can consider whether you want to add it to your boilerplate as perhaps a branch for others.

**Comment by mtompkins** [comment]
I've forked and created a new branch with the addition of the 2nd renderer in the event your want it before I try to address the store issue once more.

https://github.com/mtompkins/electron-react-boilerplate/tree/second-renderer

**Comment by mtompkins** [comment]
To the extent I'm able to tell, I believe my roadblock is the integration of the `history` module currently in the boilerplate. In its current form, I believe your use requires a DOM element which doesn't exist on the main thread. I've tried a simple change to `createMemoryHistory` but that hasn't quickly resolved anything.

I've spun up an additional branch on my fork as I work on it.
https://github.com/mtompkins/electron-react-boilerplate/tree/second-renderer-shared-store

I welcome help from anyone able.

**Comment by mtompkins** [comment]
OK - I've successfully implemented the `electron-redux` solution into your boilerplate and have made it available in a fork banch if you'd like the PR. I've added some of the comments provided from the author. I leave it to you whether it is worth cleaning up some of the `flow` and `eslint` breaks the additions have caused. I've tried hard to keep your boilerplate as original as possible.

The *gotchas* were largely `thunk` related.
